src/libexpr/primops.cc

```diff
@@ -54,6 +54,7 @@ void EvalState::realiseContext(const PathSet & context)
         auto ctx = store->parseStorePath(decoded.first);
         if (!store->isValidPath(ctx))
             throw InvalidPathError(store->printStorePath(ctx));
+        importedPaths.push_back(ctx.clone());
```

Comment:
I think we would want to push the path to the specific output rather than drv file in this case? for `--build --eval `would we get the original drv file again.

Reply:
It may be useful to have more information, but this is all that's needed for the computeFSClosure. The original drv needs to be included to get dependencies for `--build`. It may end up getting in multiple times, but I think that's okay.

From counting the number included, I am fairly sure it is the right thing. For instance, here's the deps I get for each:

<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />

<col  class="org-right" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">recursive</th>
<th scope="col" class="org-left">evaltime</th>
<th scope="col" class="org-left">buildtime</th>
<th scope="col" class="org-left">runtime</th>
<th scope="col" class="org-right">total</th>
</tr>
</thead>

<tbody>
<tr>
<td class="org-left">Y</td>
<td class="org-left">N</td>
<td class="org-left">N</td>
<td class="org-left">N</td>
<td class="org-right">0</td>
</tr>


<tr>
<td class="org-left">Y</td>
<td class="org-left">Y</td>
<td class="org-left">N</td>
<td class="org-left">N</td>
<td class="org-right">218</td>
</tr>


<tr>
<td class="org-left">Y</td>
<td class="org-left">N</td>
<td class="org-left">N</td>
<td class="org-left">Y</td>
<td class="org-right">293</td>
</tr>


<tr>
<td class="org-left">Y</td>
<td class="org-left">Y</td>
<td class="org-left">N</td>
<td class="org-left">N</td>
<td class="org-right">511</td>
</tr>


<tr>
<td class="org-left">Y</td>
<td class="org-left">N</td>
<td class="org-left">Y</td>
<td class="org-left">X</td>
<td class="org-right">1003</td>
</tr>


<tr>
<td class="org-left">Y</td>
<td class="org-left">Y</td>
<td class="org-left">Y</td>
<td class="org-left">X</td>
<td class="org-right">1008</td>
</tr>
</tbody>
</table>

Reply:
What standard commands or flags would enumerate all of these, but it can be unclear how to get a report/listing of each row. How was this generated?